test/e2e/tests/test_subscription.py

```
@@ -1210,6 +1210,125 @@ def test_rate_limit_exhaustion_gets_429(self):
+            assert r_models.status_code == 200, \
+                f"Expected 200 for /v1/models endpoint even when quota exhausted, got {r_models.status_code}. " \
+                f"The /v1/models endpoint does not consume tokens and should remain accessible. " \
+                f"Response: {r_models.text[:500]}"
+
+            # Verify it returns valid model metadata (sanity check)
+            try:
+                models_data = r_models.json()
+                assert "data" in models_data or "object" in models_data, \
+                    f"Expected valid models response with 'data' or 'object' field, got: {models_data}"
+                log.info(f"✓ /v1/models endpoint accessible (200) despite exhausted quota. Response keys: {list(models_data.keys())}")
+            except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)